### PR TITLE
Bug 1291532 - "zh_CN" should be the only parameter to the value of "I…

### PR DESCRIPTION
**Pull request by lemonYLX**
zh_CN" should be the only parameter to the value of "IsChinaEdition"

### CODE REVIEW

File: Client/Frontend/Settings/AppSettingsTableViewController.swift [hunk 1]
```diff
@@ -56,8 +56,7 @@ class AppSettingsTableViewController: SettingsTableViewController {
         ]
 
         let accountChinaSyncSetting: [Setting]
-        let locale = NSLocale.currentLocale()
-        if locale.localeIdentifier != "zh_CN" {
+        if !(profile as! BrowserProfile).isChinaEdition {
```
Comment (jhugman) [inline Client/Frontend/Settings/AppSettingsTableViewController.swift]: I'm not wild about this construction. Explicit casting may work, but will fail drastically if you try and test the app with a MockProfile. 

Suggest either:
 1. Move `isChinaEdition` into the Profile protocol, and provide a dummy implementation in `MockProfile` OR
 2. Explicitly test with a `if let` to check if it is safe to down cast to `BrowserProfile`.

Comment (lemonYLX) [inline Client/Frontend/Settings/AppSettingsTableViewController.swift]: I have moved isChinaEdition into the Profile protocol, and provided a dummy implementation in MockProfile.
What do you think of this change?

File: Client/Frontend/Settings/AppSettingsTableViewController.swift [hunk 2]
```diff
@@ -56,8 +56,7 @@ class AppSettingsTableViewController: SettingsTableViewController {
         ]
 
         let accountChinaSyncSetting: [Setting]
-        let locale = NSLocale.currentLocale()
-        if locale.localeIdentifier != "zh_CN" {
+        if  !profile.isChinaEdition {
```
Comment (jhugman) [inline Client/Frontend/Settings/AppSettingsTableViewController.swift]: Nit: whitespace.

Comment (lemonYLX) [inline Client/Frontend/Settings/AppSettingsTableViewController.swift]: Fixed.

**Pull request merged**